src/proxy/reconnect.rs

```diff
@@ -125,19 +179,35 @@ where
     type Future = ResponseFuture<N>;
 
     fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        match  self.backoff {
+            Backoff::None => {}
+            Backoff::Fixed(_) => {
+                if let Some(delay) = self.active_backoff.as_mut() {
+                    match delay.poll() {
+                        Ok(Async::NotReady) => return Ok(Async::NotReady),
+                        Ok(Async::Ready(())) => {},
+                        Err(e) => {
+                            error!("timer failed; continuing without backoff: {}", e);
+                        }
+                    }
+                }
+            }
+        };
+        self.active_backoff = None;
```

Comment:
This could go up in the match arm of `Backoff::Fixed(_)`, so it doesn't need to write if it's `None`. Something something caches...